wasmtime-runtime/Cargo.toml

```diff
@@ -33,6 +39,7 @@ cc = "1.0"
 [features]
 default = ["std"]
 std = ["cranelift-codegen/std", "cranelift-wasm/std", "wasmtime-environ/std"]
+jitdump = []
```

Comment:
Did you intend to make the jitdump functionality disabled by default, and enabled with this feature? If so, we should make the dependencies like goblin, serde, scroll, etc. optional, and enabled by the jitdump feature; see [here](https://doc.rust-lang.org/cargo/reference/manifest.html#the-features-section).

Reply:
Hi .. thanks for the suggestion. Yes, I think originally I was thinking so (thinking for debug builds only). I never really followed-up on the implementation though since a flag was guarding the dump and I wondered if a feature was really a good idea considering someone may not realize that they need to compile with the feature enabled in order for the flag to be valid. Do you have a preference? Perhaps make it a feature that is enabled by default?

Reply:
On second thought trying to make this a conditional feature is requiring guards in many places. I was trying to do something like:

> #[cfg(feature = "jitdump")]
> use crate::jit_dump::JitDumpAgent;
> #[cfg(not(feature = "jitdump"))]
> pub struct JitDumpAgent {}

In the few files that reference JitDumpAgent, but seems it is not this straight forward. Having this as a feature is leading to many build errors that are requiring more logic than expected. For now, I'll remove the code that implies this is a feature. Certainly we can make this a feature  if needed ... just perhaps could use a suggestion on the best way to do that in clean and neat way.